edx_exams/apps/lti/views.py

```diff
@@ -101,14 +101,111 @@ def acs(request, lti_config_id):
         return Response(status=400)
 
     if action == 'flag':
+        # NOTE: The flag action is not yet supported.
+        # TODO: Make the flag action actually modify the exam attempt data (or perhaps another model?)
         success_msg = (
-            f'Flagging exam for user with id {anonymous_user_id} '
+            f'NOTE: The flag action is not yet supported. The following is a placeholder message.'
+            f'Flagging exam attempt for user with id {anonymous_user_id} '
             f'with resource id {resource_id} and attempt number {attempt_number} '
             f'for lti config id {lti_config_id}, status {attempt.status}, exam id {attempt.exam.id}, '
             f'and attempt id {attempt.id}.'
         )
         log.info(success_msg)
-    return Response(status=200)
+
+    elif action == 'terminate':
+        # Upon receiving a terminate request, the attempt referenced should have their status updated
+        # depending on the reason for termination (reason_code) and the incident_severity (scaling from 0.0 to 1.0).
+        # If the severity is greater than 0.25, then the attempt is marked for secondary review.
+
+        # Get the reason code for the termination and ensure it's a string to comply with the LTI specs
+        # See: http://www.imsglobal.org/spec/proctoring/v1p0#h.rsq8h6qxveab
+        reason_code_key = data['reason_code']
+        if not isinstance(reason_code_key, str):
+            error_msg = 'ERROR: reason_code must be passed to the ACS endpoint as a string per LTI specs.'
+            error = {'detail': error_msg}
+            return Response(status=status.HTTP_400_BAD_REQUEST, data=error)
+
+        # Get the incident severity and ensure it's a string to comply with the LTI specs
+        severity = data['incident_severity']
+        if not isinstance(severity, str):
+            error_msg = 'ERROR: incident_severity must be passed to the ACS endpoint as a string per LTI specs.'
+            error = {'detail': error_msg}
+            return Response(status=status.HTTP_400_BAD_REQUEST, data=error)
+
+        # Convert reason code and severity to correct formats
+        # NOTE: Is there a point in checking if severity is a string if we're just converting it into a
+        # float anyways?
+        reason_code = ASSESSMENT_CONTROL_CODES[reason_code_key]
+        severity = float(severity)
+        SEVERITY_THRESHOLD = 0.25
+        success_msg = ''
+
+        # Reason codes:
+        # Regular submission occurred
+        # Code 1
+        if reason_code == 'Submitted by user':
```

Comment:
I think we can simplify this down to '1' means success and anything else is an error. The strings in that list are basically UI messaging so they could change and break this function. In any case checking against "magic strings" is usually not a great idea.

I don't think there's any distinction in behavior for the various errors on our end, it's mostly just information for the instructor. A non '1' response should always result in the error state regardless of severity. The instructor can figure out if it's a pass/fail/retake from there.

Reply:
side note maybe we can request they swap the 0 and 1 error codes. Would make a lot more sense since that's already a convention in exit codes.